lib/spack/spack/test/concretize.py

```diff
@@ -2093,7 +2094,25 @@ def test_result_specs_is_not_empty(self, specs):
         result, _, _ = solver.driver.solve(setup, specs, reuse=[])
 
         assert result.specs
-        assert not result.unsolved_specs
+
+    def test_unsolved_specs_raises_error(self, monkeypatch, mock_packages, config):
```

Comment:
```suggestion
    @pytest.mark.regression("38664")
    def test_unsolved_specs_raises_error(self, monkeypatch, mock_packages, config):
```

Reply:
This doesn't actually fix 38664, but I might misunderstand what we use `mark.regression` for: I do like connecting this test to that conversation so I'll update this.

Reply:
Done